packages/x-tree-view/src/internals/plugins/useTreeViewJSXNodes/useTreeViewJSXNodes.tsx

```diff
@@ -116,25 +124,22 @@ const useTreeViewJSXNodesItemPlugin: TreeViewItemPlugin = ({ props, ref }) => {
     if (label) {
       return instance.mapFirstCharFromJSX(
         nodeId,
-        (contentRef.current?.textContent ?? '').substring(0, 1).toLowerCase(),
+        (pluginContentRef.current?.textContent ?? '').substring(0, 1).toLowerCase(),
       );
     }
     return undefined;
   }, [instance, nodeId, label]);
 
   return {
-    props: {
```

Comment:
I refactored this part so that it's not aware of the props structure of the components.
Because the old `TreeItem` has its content ref in `props.ContentProps.contentRef` and the new one has its content ref in `slotProps.content.ref` 

So now the plugin just returns the ref and the item component is responsible for merging.

It means that plugins can no longer process the other props, but I think we can re-evaluate once we have an actual need and not over engineer.